src/Type/WPConnectionType.php

```diff
@@ -150,10 +150,19 @@ public function __construct( array $config, TypeRegistry $type_registry ) {
 
 		$this->validate_config( $config );
 
-		$this->config          = $config;
-		$this->from_type       = $config['fromType'];
-		$this->to_type         = $config['toType'];
-		$this->from_field_name = $config['fromFieldName'];
+		$this->config    = $config;
+		$this->from_type = $config['fromType'];
+		$this->to_type   = $config['toType'];
+
+		/**
+		 * Filter the connection field name.
+		 *
+		 * @internal This filter is internal and used by rename_graphql_field(). It is not intended for use by external code.
+		 *
+		 * @param string $from_field_name The name of the field the connection will be exposed as.
+		 */
+		$this->from_field_name = apply_filters( "graphql_wp_connection_{$this->from_type}_from_field_name", $config['fromFieldName'] );
```

Comment:
@justlevine my first thought was that this should pass more context, as it would be difficult to use in just about any way other than the current use case. 

But I think marking it internal like you've done works well. 

If/when a use case arises where a 3rd party might need to filter this, we can consider passing more context to the filter at that time.